src/format/write.rs

```diff
@@ -81,10 +81,33 @@ impl Write for &mut [u8] {
 #[cfg(feature = "alloc")]
 #[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
 impl Write for Vec<u8> {
-    fn write(&mut self, buf: &[u8]) -> Result<usize, Error> {
-        self.try_reserve(buf.len())?;
-        self.extend_from_slice(buf);
-        Ok(buf.len())
+    fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
+        self.try_reserve(data.len())?;
+        self.extend_from_slice(data);
+        Ok(data.len())
+    }
+}
+
+/// Wrapper for a [`std::io::Write`] writer.
+#[cfg(feature = "std")]
+#[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+pub(crate) struct IoWrite<'a> {
+    /// Inner writer.
+    inner: &'a mut dyn std::io::Write,
+}
+
+#[cfg(feature = "std")]
+impl<'a> IoWrite<'a> {
+    pub(crate) fn new(inner: &'a mut dyn std::io::Write) -> Self {
+        Self { inner }
+    }
+}
+
+/// Write is implemented for `IoWrite` by writing to its inner writer.
+#[cfg(feature = "std")]
+impl Write for IoWrite<'_> {
+    fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
+        Ok(self.inner.write(data)?)
```

Comment:
```suggestion
        Ok(self.inner.write_all(data)?)
```

Reply:
Do we need a test for this bug?

Reply:
This is not a bug, we need to keep the `write` call here. We also may want to override the default `write_all` method to call the inner `write_all` directly.